Formula/jetty.rb

```diff
@@ -23,28 +23,31 @@ def install
     libexec.install Dir["*"]
     (libexec/"logs").mkpath
 
-    bin.mkpath
-    Dir.glob("#{libexec}/bin/*.sh") do |f|
-      scriptname = File.basename(f, ".sh")
-      (bin/scriptname).write <<~EOS
-        #!/bin/bash
-        export JETTY_HOME='#{libexec}'
-        export JAVA_HOME="${JAVA_HOME:-#{Formula["openjdk"].opt_prefix}}"
-        exec #{f} "$@"
-      EOS
-      chmod 0755, bin/scriptname
+    env = Language::Java.overridable_java_home_env
+    env["JETTY_HOME"] = libexec
+    Dir.glob(libexec/"bin/*.sh") do |f|
+      (bin/File.basename(f, ".sh")).write_env_script f, env
     end
   end
 
   test do
-    ENV["JETTY_ARGS"] = "jetty.http.port=#{free_port} jetty.ssl.port=#{free_port}"
+    http_port = free_port
+    ENV["JETTY_ARGS"] = "jetty.http.port=#{http_port} jetty.ssl.port=#{free_port}"
     ENV["JETTY_BASE"] = testpath
-    cp_r Dir[libexec/"*"], testpath
-    pid = fork { exec bin/"jetty", "start" }
-    sleep 5 # grace time for server start
+    ENV["JETTY_RUN"] = testpath
+    cp_r Dir[libexec/"demo-base/*"], testpath
+
+    log = testpath/"jetty.log"
+    pid = fork do
+      $stdout.reopen(log)
+      $stderr.reopen(log)
+      exec bin/"jetty", "run"
+    end
+
     begin
-      assert_match(/Jetty running pid=\d+/, shell_output("#{bin}/jetty check"))
-      assert_equal "Stopping Jetty: OK\n", shell_output("#{bin}/jetty stop")
```

Comment:
`jetty run` keeps process in foreground but doesn't work with `jetty check` and other commands since it doesn't write PID file. It does help minimize issues as it doesn't orphan the process unlike `jetty start` (which runs `disown`).

Also changed test to run an actual example (demo-base), which allows `curl`'ing local HTTP server.